With kernel-3.2.57, ip_select_ident first agrument changed from "struct iphdr *iph" to struct sk_buff *skb. Changing the usage in vxlan_driver to reflect the same.

diff --git a/drivers/net/vxlan.c b/drivers/net/vxlan.c
index 2fdd4e8..d4ce65e 100644
--- a/drivers/net/vxlan.c
+++ b/drivers/net/vxlan.c
@@ -813,7 +813,7 @@ static netdev_tx_t vxlan_xmit_one(struct sk_buff *skb, struct net_device *dev, _
 
 	/* See __IPTUNNEL_XMIT */
 	skb->ip_summed = CHECKSUM_NONE;
-	ip_select_ident(iph, &rt->dst, NULL);
+	ip_select_ident(skb, &rt->dst, NULL);
 
 	err = ip_local_out(skb);
 	if (likely(net_xmit_eval(err) == 0)) {
